Docker build fails to install osm2pgsql - NLOHMANN_INCLUDE_DIR

## What version of PgOSM Flex are you using?

Latest feature branch, re #344.

## Docker image

GitHub is failing to build the Docker image.  


## What did you do exactly?

Pushed changes to GitHub.


## What did you expect to happen?

Actions should successfully build the Docker image.

## What did happen instead?

GitHub actions fails to build, first failure: https://github.com/rustprooflabs/pgosm-flex/actions/runs/5448223334

Error message pointing to what appears to be the problem.

```bash
#13 2.224 CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
#13 2.224 Please set them or make sure they are set and tested correctly in the CMake files:
#13 2.224 NLOHMANN_INCLUDE_DIR
```



## What did you do to try analyzing the problem?

Searching for NLOHMANN_INCLUDE_DIR led me to `CMakeLists.txt` in osm2pgsql.  Git blame [on that line](https://github.com/openstreetmap/osm2pgsql/blame/8714b14d1a827b0624d56a561451c19147716fcc/CMakeLists.txt#L204
) points to a [commit](https://github.com/openstreetmap/osm2pgsql/commit/70b792b4260cf37bdd89f6aeb79e7425fba1a7c0#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a) from about 12 hours earlier.  Related PR https://github.com/openstreetmap/osm2pgsql/pull/1969

It appears there's a new dependency required to be installed to support this change.  It hasn't failed for me locally yet since I've been using docker build using the cache with a slightly older version of osm2pgsql.

--- a/Dockerfile
+++ b/Dockerfile
@@ -17,6 +17,7 @@ RUN apt-get update \
         postgresql-server-dev-15 \
         curl unzip \
         postgresql-15-pgrouting \
+        nlohmann-json3-dev \
     && rm -rf /var/lib/apt/lists/*
 
 RUN wget https://luarocks.org/releases/luarocks-3.9.1.tar.gz \

